tests/test_reserved_terms.py

```diff
@@ -0,0 +1,17 @@
+from nomenclature import DataStructureDefinition
+
+
+def test_variable_ops_as_square_brackets():
```

Comment:
Should this not go into nomenclature as part of `VariableCodeList`? I think we would want to enforce these standards also outside of common definitions going forward. Maybe with the option to suppress the error for legacy variable names.